src/providers/drafts.ts

```diff
@@ -0,0 +1,140 @@
+  category?: VideoCategory['id'] | null
+  licenseCode?: License['code'] | null
+  licenseCustomText?: License['customText'] | null
+  licenseAttribution?: License['attribution'] | null
+} & CommonDraftProps
+
+export type RawDraft = Omit<Draft, 'id' | 'updatedAt'>
+interface DraftStoreStateV1 {
+  allDrafts: Draft[]
+  allUnseenDrafts: Array<{
+    draftId: string
+    channelId: string
+  }>
+}
+interface DraftStoreStateV2 {
+  allDrafts: Draft[]
+}
+interface DraftStoreActions {
+  addDraft: (draft: RawDraft, explicitId?: string) => Draft
+  updateDraft: (draftId: string, draftProps: RawDraft) => void
+  getDraft: (draftId: string) => Draft | undefined
+  getDraftsForChannel: (draftIds: string) => Draft[]
+  getUnseenDraftsForChannel: (draftIds: string) => Draft[]
+  markAllDraftsAsSeenForChannel: (channelId: string) => void
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreStateV2, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
+    },
+    actionsFactory: (set, get) => ({
+      addDraft: (draft, explicitId) => {
+        const id = explicitId ?? createId()
+        const updatedAt = new Date().toISOString()
+        const newDraft: Draft = { ...draft, updatedAt, id, seen: false }
+        set((draftState) => {
+          draftState.allDrafts = [newDraft, ...draftState.allDrafts]
+        })
+        return newDraft
+      },
+      updateDraft: (draftId, draftProps) => {
+        const updatedAt = new Date().toISOString()
+        set((draftState) => {
+          const idx = draftState.allDrafts.findIndex((d) => d.id === draftId)
+          if (idx >= 0) {
+            draftState.allDrafts[idx] = { ...draftState.allDrafts[idx], ...draftProps, updatedAt }
+          }
+        })
+      },
+      getDraft: (id) => get().allDrafts.find((draft) => draft.id === id),
+      getDraftsForChannel: (channelId) => get().allDrafts.filter((d) => d.channelId === channelId),
+      getUnseenDraftsForChannel: (channelId) =>
+        get().allDrafts.filter((d) => d.channelId === channelId && d.seen === false),
+      removeDrafts: (draftIds) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => !draftIds.includes(draft.id))
+        })
+      },
+      removeAllDrafts: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => draft.channelId !== channelId)
+        })
+      },
+      markAllDraftsAsSeenForChannel: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.map((draft) => ({
+            ...draft,
+            seen: draft.channelId === channelId ? true : draft.seen,
+          }))
+        })
+      },
+    }),
+  },
+  {
+    persist: {
+      key: 'drafts',
+      whitelist: ['allDrafts'],
+      version: 2,
```

Comment:
We only had version `1` for this PR right? Since this wasn't ever deployed, I don't think we need to increment to `2` and migrate

Reply:
True !!